src/Nethermind/Nethermind.Network/NetworkStorage.cs

```diff
@@ -139,17 +135,10 @@ private void RemoveLocal(PublicKey nodeId)
         {
             lock (_lock)
             {
-                Span<NetworkNode> span = CollectionsMarshal.AsSpan(_nodesList);
-                for (int i = 0; i < span.Length; i++)
+                if (_nodesDict.Remove(nodeId))
```

Comment:
Would concurrent dictionary be better than all those lock's?

Reply:
Is mostly single threaded so would be a bit heavy weight